### PR TITLE
Simplify reporting server errors.

### PR DESCRIPTION
**Pull request by hluk**
The custom pretty_print wasn't able to handle some data types.
Using JSON seems to be more straightforward.

This is also required by [PDC-1917](https://projects.engineering.redhat.com/browse/PDC-1917) to correctly print result of import command.

### CODE REVIEW

**Review by hluk — COMMENTED** [review]
+1

Removing lot of stuff? I approve!

Error messages are bit different now though.

File: pdc_client/runner.py [hunk 1]
```diff
@@ -167,32 +167,13 @@ def run(self, args=None):
...
-                json = exc.response.json()
-                pretty_print(json, file=sys.stderr)
-            except ValueError:
-                # Response was not JSON
-                print('Failed to parse error response.', file=sys.stderr)
-            except TypeError as e:
-                # Failed to pretty print.
-                print('Failed to correctly display error message. Please file a bug.',
-                      file=sys.stderr)
-                self.logger.info(json, exc_info=e)
```
Comment (xiangge) [inline pdc_client/runner.py:183]: When I try to use a wrong command to get information.

Just execute two times of the command, when I execute the second time, you will get below error with many html codes, I think it is not way to read.
So I think we should keep the ValueError and TypeError parts, what's your thinking @hluk ?
I get below:
```
(pdc) [chcao@dhcp-128-226 pdc_client]$ pdc -s local group-resource-permissions create --group devel --resource arches --permission read
Server returned following error: [500] INTERNAL SERVER ERROR
Details:

<!DOCTYPE html>
<html lang="en">
<head>
  <meta http-equiv="content-type" content="text/html; charset=utf-8">
  <meta name="robots" content="NONE,NOARCHIVE">
  <title>FieldError at /rest_api/v1/auth/group-resource-permissions/</title>
  <style type="text/css">
    html * { padding:0; margin:0; }
    body * { padding:10px 20px; }
    body * * { padding:0; }
    body { font:small sans-serif; }
    body>div { border-bottom:1px solid #ddd; }
    h1 { font-weight:normal; }
    h2 { margin-bottom:.8em; }
    h2 span { font-size:80%; color:#666; font-weight:normal; }
    h3 { margin:1em 0 .5em 0; }
    h4 { margin:0 0 .5em 0; font-weight: normal; }
    code, pre { font-size: 100%; white-space: pre-wrap; }
    table { border:1px solid #ccc; border-collapse: collapse; width:100%; background:white; }
    tbody td, tbody th { vertical-align:top; padding:2px 3px; }
    thead th {
      padding:1px 6px 1px 3px; background:#fefefe; text-align:left;
......
```

Comment (hluk) [reply pdc_client/runner.py:183]: OK, fixing it, thanks for the info.

Comment (hluk) [reply pdc_client/runner.py:183]: Fixed. Now it prints:

```
Server returned following error: [500] INTERNAL SERVER ERROR
Details:
Failed to parse error response.
```

BTW, for that particular `pdc` command, I think server should send JSON with the error. Looks like uncaught exception on server:

```
FieldError: ['The fields resource, permission, group must make a unique set.']
```

**Comment by hluk** [comment]
Example output of an error:

```
> pdc -s local group-resource-permissions create --group test_group --resource arches --permission read
Server returned following error: [400] BAD REQUEST
Details:
{
  "detail": [
    "The fields resource, permission, group must make a unique set."
  ]
} 
```

I think this should print just the detail text.

File: pdc_client/runner.py (hunk 1 above)
Comment (xiangge) [reply pdc_client/runner.py:183]: Yes, looks good to me.

**Comment by xiangge** [comment]
Agree with your point.

**Comment by simozhan** [comment]
:+1:  Actually i like the previous indication for the end user "print('Internal server error. Please consider reporting a bug.'", but yeah, this is maybe a minor thing.

**Comment by hluk** [comment]
The errors should be now less verbose. Is it OK to merge?

**Review by simozhan — APPROVED** [review]
Yeah. Ok for merge

**Review by xiangge — APPROVED** [review]

**Pull request merged**